### ISSUE TITLE
Static analysis: Overflowed return value

### ISSUE THREAD

**Issue by tbeu** [weiss/c99-snprintf#1]
Coverity Scan reports an Overflowed return value from function `myround`.

```c
static UINTMAX_T
myround(LDOUBLE value)
{
    UINTMAX_T intpart = cast(value);

//1. Condition (value -= intpart) < 0.5, taking false branch.
//2. overflow: Add operation overflows on operands intpart and 1UL.
//CID 59973 (#1 of 1): Overflowed return value (INTEGER_OVERFLOW)
//3. overflow_sink: Overflowed or truncated value (or a value computed from an overflowed or truncated value) ((value -= intpart) < 0.5) ? intpart : (intpart + 1UL) used as return value.
    return ((value -= intpart) < 0.5) ? intpart : intpart + 1;
}
```

**Comment by zfields** [comment]
The call to `cast()` can return `UINTMAX_MAX`, in which case the return statement will overflow.

**Comment by tbeu** [comment]
This can be closed after merging #10 .

**Comment by weiss** [comment]
Thanks 👍🏻
